posthog/api/notebook.py

```diff
@@ -345,6 +347,40 @@ def retrieve(self, request: Request, *args: Any, **kwargs: Any) -> Response:
 
         return Response(serializer.data)
 
+    @action(methods=["GET"], detail=False)
+    def recording_comments(self, request: Request, **kwargs):
+        recording_id = request.GET.get("recording_id")
+        if not recording_id:
+            return Response({"detail": "recording_id is required"}, status=400)
+
+        queryset = self.safely_get_queryset(self.queryset)
+        queryset = self._filter_list_request(request, queryset, {"contains": f"recording:{recording_id}"})
+        notebooks = queryset.all()
+        comments = []
+        for notebook in notebooks:
+            content_nodes = notebook.content.get("content", {})
+            for node in content_nodes:
+                if node.get("type", None) == "paragraph" and len(node.get("content", [])) == 2:
+                    attrs = node.get("content", [])[0].get("attrs", {})
+                    content_node_recording_id = attrs.get("sessionRecordingId", None)
+                    playback_time = attrs.get("playbackTime", None)
+                    text = node.get("content", [])[1].get("text", None)
```

Comment:
Probably not an expensive operation but this could be moved inside the conditional

Reply:
agree likely cheap... but safe to move and protects us for very large notebooks so 👍